iam_aws_vpc_cni.tf

```
@@ -1,5 +1,5 @@
 locals {
-  handle_aws_vpc_cni = var.handle_iam_resources && (var.handle_iam_aws_vpc_cni || contains(keys(var.cluster_addons), "vpc-cni"))
+  handle_aws_vpc_cni = var.handle_iam_resources && (var.handle_iam_aws_vpc_cni || length(var.vpc_cni_addon) > 0)
```

Why do you test length(var.vpc_cni_addon) > 0) ? 
```suggestion
  handle_aws_vpc_cni = var.handle_iam_resources && var.handle_iam_aws_vpc_cni
```

---

You are right, this can be tested in the creation of the vpc-cni addon itself